src/opencmw_worker/src/NilmDataWorker.hpp

```diff
@@ -0,0 +1,208 @@
+#ifndef NILM_DATA_WORKER_H
+#define NILM_DATA_WORKER_H
+
+#include <disruptor/RingBuffer.hpp>
+#include <majordomo/Worker.hpp>
+
+#include <gnuradio/pulsed_power/opencmw_freq_sink.h>
+#include <gnuradio/pulsed_power/opencmw_time_sink.h>
+
+using opencmw::Annotated;
+using opencmw::NoUnit;
+
+// context for InferenceTool
+struct NilmAcquisitionContext {
+    int64_t                 lastRefTrigger = 0;
+    opencmw::MIME::MimeType contentType    = opencmw::MIME::JSON;
+};
+
+ENABLE_REFLECTION_FOR(NilmAcquisitionContext, lastRefTrigger, contentType)
+
+// data for InferenceTool
+struct AcquisitionNilm {
+    std::vector<int64_t> refTriggerStamp;
+    std::vector<float>   apparentPowerSpectrumStridedValues; // fft(S)
+    std::vector<float>   realPower;                          // P
+    std::vector<float>   reactivePower;                      // Q
+    std::vector<float>   apparentPower;                      // S
+    std::vector<float>   phi;
+};
```

Comment:
> Why is P, Q, S, phi delivered via the AcquisitionNilm subscription?

Because I wanted to avoid an additional HTTP request and deliver all the required data in one request-response. Since the spectrums are already quite big to transfer over the network, I did not want to cause more delays.

> Or do you need to have it in one object in order to synchronize the power spectra in time?

Yes, it is now one object and the synchronization is done in NilmDataWorker.